src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -518,10 +518,10 @@ void BlockchainLMDB::do_resize(uint64_t increase_size)
   const uint64_t add_size = 1LL << 30;
 
   // check disk capacity
+  boost::filesystem::path path(m_folder);
+  boost::filesystem::space_info si = boost::filesystem::space(path);
```

Comment:
What about the try/catch block?

Reply:
there `si` redeclared

Reply:
boost::filesystem::space can still fail

Reply:
oh got it now

Reply:
How about this boost's magic potion @selsta ? (i think it can be used instead of that try/catch block as well)

Reply:
This does not look correct to me. si.available is uintmax_t and not an optional type. This does not catch filesystem_error exceptions.

Reply:
ok you most likely misunderstand the use of boost::optional 's class template.
_DESCRIPTION:
The class template boost::optional manages an optional contained value (a value that or may not be present).A common use of boost::optional is in cases when the return value of a function may fail in some cases._

Reply:
I know the use case of boost optional but this requires a boost::optional return value which is not the case here. See here as an example:

https://www.boost.org/doc/libs/1_66_0/libs/optional/doc/html/boost_optional/quick_start.html

But I will leave this to the more experienced devs :P

Reply:
Actually boost::option maybe used for any nullable object see this example https://www.boost.org/doc/libs/1_65_1/libs/optional/doc/html/index.html
I am closing this and leave more experienced devs :P to keep on suggesting code to your project

Reply:
Your example also has an boost::optional return value: `boost::optional<int> getConfigParam`

> I am closing this and leave more experienced devs :P to keep on suggesting code to your project

Not sure why you closed this now. I only said that I will leave this review to devs who are more experienced than me.

Reply:
Pardon me for being jumpy.

Reply:
selsta is right. Assigning a return value will happen if no exception is thrown by the call. If an exception is thrown, that assignment will be skipped anyway. The optional is pointless in this case. Use try/catch and get rid of the optional. Using optional would only make sense for an API which was designed to use it, which this seems not to.